assembly/src/ast/parsers/constants.rs

```diff
@@ -0,0 +1,262 @@
+) -> Result<Felt, ParsingError> {
+    let mut stack = Vec::new();
+
+    for operation in postfix_expression.iter() {
+        match operation {
+            // if the operation is a value
+            Operation::Value(value) => stack.push(*value),
+            // if the operation is an operator
+            _ => {
+                let right = stack.pop().expect("stack is empty");
+                let left = stack.pop().expect("stack is empty");
+                stack.push(compute_statement(left, right, operation, op, expression)?);
+            }
+        }
+    }
+
+    // get the result from the stack
+    stack.pop().ok_or_else(|| {
+        ParsingError::invalid_const_value(
+            op,
+            expression,
+            &format!("constant expression {} is incorrect", op),
+        )
+    })
+}
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Returns the number in `value` or the constant value if the value is the name of the constant.
+fn parse_number(
+    op: &Token,
+    expression: &str,
+    constants: &LocalConstMap,
+    value: String,
+) -> Result<Operation, ParsingError> {
+    let parsed_number = value.parse::<u64>();
+    // if the parsed value is a number push it on the stack
+    if let Ok(parsed_number) = parsed_number {
+        Ok(Operation::Value(Felt::new(parsed_number)))
+    }
+    // if it is a name of the constant get its value from the `constants` map
+    else {
+        let parsed_number = constants.get(&value).ok_or_else(|| {
+            ParsingError::invalid_const_value(
+                op,
+                expression,
+                &format!("constant with name {} was not initialized", value),
+            )
+        })?;
+        Ok(Operation::Value(Felt::new(*parsed_number)))
+    }
+}
+
+/// Returns `true` if th left operator has higher priority than the right, `false` otherwise.
+fn left_has_greater_priority(left: &char, right: &Operation) -> bool {
+    use Operation::*;
+
+    let left_level = match left {
+        '*' | '/' => 2,
+        '+' | '-' => 1,
+        _ => 0,
+    };
+    let right_level = match right {
+        Mul | Div | IntDiv => 2,
+        Add | Sub => 1,
+        _ => 0,
+    };
+    left_level > right_level
+}
+
+/// Computes the expression based on provided `operator` character.
+fn compute_statement(
+    left: Felt,
+    right: Felt,
+    operator: &Operation,
+    op: &Token,
+    expression: &str,
+) -> Result<Felt, ParsingError> {
+    use Operation::*;
+    match operator {
+        Add => Ok(left + right),
+        Sub => Ok(left - right),
+        Mul => Ok(left * right),
+        Div => Ok(Felt::new(left.as_int() / right.as_int())),
+        IntDiv => Ok(left / right),
+        _ => Err(ParsingError::invalid_const_value(
+            op,
+            expression,
+            &format!("expression contains unsupported operator: {}", operator),
+        )),
```

Comment:
Under what circumstances could this error happen? I think we should be able to check for all errors while parsing the infix expression. If so, then this can be just `unreachable!()`.